tests/claude_code/test_config.py

```diff
@@ -0,0 +1,202 @@
+    claude_settings_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(claude_settings_path, "w") as f:
+        json.dump(config_data, f)
+
+    # Change to the temp directory so .claude/settings.json is found
+    original_cwd = os.getcwd()
+    try:
+        os.chdir(str(tmp_path))
+        result = get_env_var(MLFLOW_TRACING_ENABLED, "default")
+        assert result == "claude_value"
+    finally:
+        os.chdir(original_cwd)
+
+
+def test_get_env_var_default_when_not_found(tmp_path, monkeypatch):
+    """Test get_env_var returns default when variable not found anywhere."""
+    # Ensure OS env var is not set
+    monkeypatch.delenv(MLFLOW_TRACING_ENABLED, raising=False)
+
+    # Create empty settings file in .claude directory
+    claude_settings_path = tmp_path / ".claude" / "settings.json"
+    claude_settings_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(claude_settings_path, "w") as f:
+        json.dump({}, f)
+
+    # Change to temp directory so .claude/settings.json is found
+    original_cwd = os.getcwd()
+    try:
+        os.chdir(str(tmp_path))
+        result = get_env_var(MLFLOW_TRACING_ENABLED, "default_value")
+        assert result == "default_value"
+    finally:
+        os.chdir(original_cwd)
+
+
+def test_get_tracing_status_enabled(temp_settings_path):
+    """Test get_tracing_status returns enabled status when tracing is enabled."""
+    # Create settings with tracing enabled
+    config_data = {"environment": {MLFLOW_TRACING_ENABLED: "true"}}
+    with open(temp_settings_path, "w") as f:
+        json.dump(config_data, f)
+
+    status = get_tracing_status(temp_settings_path)
+    assert status["enabled"] is True
+    assert "tracking_uri" in status
+
+
+def test_get_tracing_status_disabled(temp_settings_path):
+    """Test get_tracing_status returns disabled status when tracing is disabled."""
+    # Create settings with tracing disabled
+    config_data = {"environment": {MLFLOW_TRACING_ENABLED: "false"}}
+    with open(temp_settings_path, "w") as f:
+        json.dump(config_data, f)
+
+    status = get_tracing_status(temp_settings_path)
+    assert status["enabled"] is False
+
+
+def test_get_tracing_status_no_config(tmp_path):
+    """Test get_tracing_status returns disabled when no configuration exists."""
+    non_existent_path = tmp_path / "missing.json"
+    status = get_tracing_status(non_existent_path)
+    assert status["enabled"] is False
+    assert status["reason"] == "No configuration found"
+
+
+def test_setup_environment_config_new_file(temp_settings_path):
+    """Test setup_environment_config creates new configuration file."""
+    tracking_uri = "test://localhost"
+    experiment_id = "123"
+
+    setup_environment_config(temp_settings_path, tracking_uri, experiment_id)
+
+    # Verify file was created
+    assert temp_settings_path.exists()
+
+    # Verify configuration contents
+    with open(temp_settings_path) as f:
+        config = json.load(f)
+
```

Comment:
```suggestion
    config = json.loads(temp_settings_path.read_text())
```

not a blocker, but we can make this one liner.